modular_zubbers/modules/plexagon_selfserve/code/plexagon_selfserve.dm

```diff
@@ -0,0 +1,282 @@
+/datum/computer_file/program/crew_self_serve
+	filename = "plexagoncrewlogin"
+	filedesc = "Plexagon Punch Clock"
+	downloader_category = PROGRAM_CATEGORY_SECURITY
+	program_open_overlay = "generic"
+	extended_desc = "Allows crew members to remotely clock in or clock out to their jobs."
+	program_flags = PROGRAM_ON_NTNET_STORE
+	size = 4
+	tgui_id = "NtosSelfServe"
+	program_icon = "id-card"
+	/// The ID card used to authenticate.
+	var/obj/item/card/id/authenticated_card
+	/// The name of the registered user, related to `authenticated_card`.
+	var/authenticated_user
+	///What trim is applied to inserted IDs?
+	var/target_trim = /datum/id_trim/job/assistant
+
+
+/**
+ * Authenticates the program based on the specific ID card.
+ *
+ * Arguments:
+ * * auth_card - The ID card to attempt to authenticate under.
+ */
+/datum/computer_file/program/crew_self_serve/proc/authenticate(obj/item/card/id/id_card)
+	if(!id_card)
+		authenticated_card = null
+		authenticated_user = null
+		return FALSE
+	else
+		authenticated_card = id_card
+		authenticated_user = "[authenticated_card.name]"
+		return TRUE
+
+/datum/computer_file/program/crew_self_serve/on_start(mob/living/user)
+	. = ..()
+	if(!.)
+		return FALSE
+
+	if(!computer)
+		to_chat(user, span_warning("Plexagon: Login failed, your computer is nonexistant!"))
+		playsound(computer, 'sound/machines/buzz-sigh.ogg', 50, FALSE)
+		return FALSE
+
+	computer.crew_manifest_update = TRUE
+	if (!computer.computer_id_slot)
+		to_chat(user, span_warning("Plexagon: Login failed, no ID card found!"))
+		playsound(computer, 'sound/machines/buzz-sigh.ogg', 50, FALSE)
+		return FALSE
+	else if(authenticate(computer.computer_id_slot))
+		return TRUE
+	else
+		return FALSE
+
+/// Clocks out the currently inserted ID Card
+/datum/computer_file/program/crew_self_serve/proc/clock_out()
+	if(!authenticated_card)
```

Comment:
I would much prefer if the clock out proc was shared by the stationary clock out computer, and the proc itself was derived from the jobs subsystem or something similar, same for clock in, there's a lot of logic that really should be standardized between the two